sympy/matrices/common.py

```diff
@@ -2305,16 +2367,17 @@ def __pow__(self, exp):
             # When certain conditions are met,
             # Jordan block algorithm is faster than
             # computation by recursion.
-            elif a.rows == 2 and exp > 100000 and jordan_pow is not None:
+            elif jordan_pow is not None and (jordan or \
+                    (jordan is not False and a.rows == 2 and exp > 100000)):
                 try:
-                    return jordan_pow(exp)
+                    return jordan_pow(exp, mulsimp=mulsimp)
                 except MatrixError:
                     pass
```

Comment:
I think maybe if `jordan_pow` is passed explicitly then the `MatrixError` should be allowed to propagate.

Reply:
Will do.